lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1186,14 +1162,37 @@ def main():
         update_max_failure_ratio=dict(default=0, type='float'),
         update_order=dict(default=None, type='str'),
         user=dict(default='root'))
+
+    option_minimal_versions = dict(
+        dns=dict(docker_py_version='2.6.0', docker_api_version='1.25'),
+        dns_options=dict(docker_py_version='2.6.0', docker_api_version='1.25'),
+        dns_search=dict(docker_py_version='2.6.0', docker_api_version='1.25'),
+        force_update=dict(docker_py_version='2.1.0', docker_api_version='1.25'),
+        hostname=dict(docker_py_version='2.2.0', docker_api_version='1.25'),
+        tty=dict(docker_py_version='2.4.0', docker_api_version='1.25'),
+        secrets=dict(docker_py_version='2.1.0', docker_api_version='1.25'),
+        configs=dict(docker_py_version='2.6.0', docker_api_version='1.30'),
+        update_order=dict(docker_py_version='2.7.0', docker_api_version='1.29'),
+        # specials
+        publish_mode=dict(
+            docker_py_version='3.0.0',
+            docker_api_version='1.25',
+            detect_usage=_detect_publish_mode_usage,
+            usage_msg='set publish.mode'
+        )
+    )
+
     required_if = [
         ('state', 'present', ['image'])
     ]
+
     client = AnsibleDockerClient(
         argument_spec=argument_spec,
         required_if=required_if,
         supports_check_mode=True,
         min_docker_version='2.0.0',
+        min_docker_api_version='1.24',
+        option_minimal_versions=option_minimal_versions
```

Comment:
While we're at it: please add a trailing comma, that makes it easier to add another option (if we ever need to) :)

Reply:
👍